diff --git a/lua/lazy/manage/task/plugin.lua b/lua/lazy/manage/task/plugin.lua
index c84e8a4..bf0fd2e 100644
--- a/lua/lazy/manage/task/plugin.lua
+++ b/lua/lazy/manage/task/plugin.lua
@@ -91,6 +91,9 @@ M.docs = {
   run = function(self)
     local docs = self.plugin.dir .. "/doc"
     if Util.file_exists(docs) then
+      if not vim.uv.fs_access(docs, "W") then
+        return
+      end
       self:log(vim.api.nvim_cmd({ cmd = "helptags", args = { docs } }, { output = true }))
     end
   end,
